src/base/hip/backend_hip.cpp

```diff
@@ -60,7 +60,14 @@ namespace rocalution
         _get_backend_descriptor()->ROC_sparse_handle = new rocsparse_handle;
 
         // get last error (if any)
-        hipGetLastError();
+        hipError_t hip_status_t = hipGetLastError();
+
+        if(hip_status_t != hipSuccess)
+        {
+            LOG_INFO("HIP error encountered during rocALUTION HIP backend initialization - falling "
+                     "back to host backend");
+            return false;
+        }
 
```

Comment:
Not sure if this is intended behaviour. I think we are calling `hipGetLastError()` in order to clear the error backlog before actually starting the backend.